test/behavior/bugs/13366.zig

```diff
@@ -0,0 +1,21 @@
+const std = @import("std");
+const expect = std.testing.expect;
+
+const ComptimeReason = union(enum) {
+    c_import: struct {
+        a: u32,
+    },
+};
+
+const Block = struct {
+    reason: ?*const ComptimeReason,
+};
+
+test {
```

Comment:
Like in my PR you basically need to skip all in-house backends as they just aren't advanced enough yet to handle most of these bugs:
https://github.com/ziglang/zig/pull/13810/files#diff-61535ab06ad68a42128cbf8f01c818fad92be29bf324739ecf0494235ea28b39R6-R10

Here's how you can test this locally:
```
zig test test/behavior/bugs/13366.zig -target wasm32-linux -fno-LLVM
```
`-fno-LLVM` is what enables the in-house backends.
If you test wasm32-linux, x86_64-linux, x86-linux, aarch64-linux, and arm-linux for the target, you'll find that they all either compile-error with some TODO or segfault.

Therefore I suggest:
```suggestion
test {
    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest; // TODO
    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest; // TODO
    if (builtin.zig_backend == .stage2_x86) return error.SkipZigTest; // TODO
    if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
    if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO

```

Reply:
Thank you! I didn't know how this aspect of zig worked so I didn't know what to make of the test failure. I'm updating it with your first suggestion now. In the future, I should be able to get it right the first time :)